modules/javafx.graphics/src/main/java/com/sun/glass/ui/mac/MacMenuDelegate.java

```diff
@@ -90,6 +90,7 @@ private native long _createMenuItem(String title, char shortcut, int modifiers,
     @Override public boolean remove(MenuItemDelegate item, int pos) {
         MacMenuDelegate macMenu = (MacMenuDelegate)item;
         _remove(ptr, macMenu == null ? 0L : macMenu.ptr, pos);
+        macMenu.ptr = 0L;
```

Comment:
I added the null check on `macMenu`.
I also added the zeroing in the other `remove` method (which indeed is for adding a menu to a menu) and to the `remove` method in `MacMenuBarDelegate`.
Apart from that, I noticed that there were a number of possible scenario's where the Delegate objects would still be used. For example, after removing a menuItem from a menu, one could still invoke methods on that menuItem object. The invocations would propagate to the native layer, causing a crash because the pointer to the `GlassMenu` was already zero (and the native GlassMenu instance was released).